core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolution.js

```diff
@@ -0,0 +1,242 @@
+    $scope.isCurrentInteractionLinear = function() {
+      var interactionId = stateInteractionIdService.savedMemento;
+      return interactionId && INTERACTION_SPECS[interactionId].is_linear;
+    };
+
+    $scope.openAddSolutionModal = function() {
+      alertsService.clearWarnings();
+      $rootScope.$broadcast('externalSave');
+      $scope.inlineSolutionEditorIsActive = false;
+
+      $modal.open({
+        templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+          '/pages/exploration_editor/editor_tab/addSolutionModal.html'),
+        backdrop: 'static',
+        resolve: {
+          objectType: function() {
+            return $scope.objectType;
+          },
+          correctAnswer: function() {
+            return $scope.correctAnswer;
+          },
+          interactionHtml: function() {
+            return $scope.interactionHtml;
+          }
+        },
+        controller: [
+          '$scope', '$modalInstance', 'objectType', 'correctAnswer',
+          'interactionHtml',
+          function($scope, $modalInstance, objectType, correctAnswer,
+            interactionHtml) {
+            $scope.correctAnswer = correctAnswer;
+            $scope.objectType = objectType;
+            $scope.interactionHtml = interactionHtml;
+            $scope.EXPLANATION_FORM_SCHEMA = {
+              type: 'html',
+              ui_config: {}
+            };
+
+            $scope.submitAnswer = function(answer) {
+              $scope.correctAnswer = answer;
+            };
+
+            $scope.tmpSolution = {};
+            $scope.tmpSolution.answerIsExclusive = false;
+            $scope.tmpSolution.correctAnswer = '';
+            $scope.tmpSolution.explanation = '';
+
+            $scope.addSolutionForm = {};
+
+            $scope.saveSolution = function() {
+              // Close the modal and save it afterwards.
+              var answer = $scope.correctAnswer;
+              if (!interactionHtml) {
+                answer = $scope.tmpSolution.correctAnswer;
+              }
+              answer = (
+                StateSolutionHelperService.getCorrectAnswerObject(answer,
+                  $scope.objectType));
+              $modalInstance.close({
+                solution: SolutionObjectFactory.createNew(
+                    $scope.tmpSolution.answerIsExclusive,
+                    answer,
+                    $scope.tmpSolution.explanation)
+              });
+            };
+
+            $scope.cancel = function() {
+              $modalInstance.dismiss('cancel');
+              alertsService.clearWarnings();
+            };
+          }
+        ]
+      }).result.then(function(result) {
+        var correctAnswer = result.solution.correctAnswer;
+        var currentStateName = editorContextService.getActiveStateName();
+        var state = explorationStatesService.getState(currentStateName);
+        try {
+          StateSolutionHelperService.verifyAndSaveAnswer(
```

Comment:
I don't see a throw in this function; can you explain what you're catching here?